drivers/spi/spi_ll_stm32.c

```diff
@@ -678,6 +709,52 @@ static int spi_stm32_release(const struct device *dev,
 	return 0;
 }
 
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_spi)
+static int32_t spi_stm32_count_bufset_frames(const struct spi_config *config,
+			const struct spi_buf_set *bufs)
+{
+	if (bufs == NULL) {
+		return 0;
+	}
+
+	uint32_t num_bytes = 0;
```

Comment:
Variables should be declared at the beginning of the block (see coding guidelines).

Reply:
Sorry, could you point me to the exact rule? I'm inspecting this coding guidelines document: https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html and I can't find the related rule.

Reply:
Ha, it seems this page has evolved since the last time I've checked. Only close reference (which would answer your exact question) I've found is https://wiki.sei.cmu.edu/confluence/display/c/DCL19-C.+Minimize+the+scope+of+variables+and+functions, but it does not specify the rule I mention.
I'll make more research

Reply:
Doesn't see to be explicitly stated anywhere, but it is a general C coding best practice, recommended by K & R: `In Chapter 1.2 of K&R is written: In C, all variables must be declared before they are used, usually at the beginning of a function, before any executable statements.`
Take this comment as non blocking.

Reply:
IMHO
Most drivers in Zephyr are written that way - "_Variables should be declared at the beginning of the block_".
It would be preferable to write new code in a similar way to existing one.